diff --git a/scripts/build-goma-dependencies.sh b/scripts/build-goma-dependencies.sh
index cff0197..f2310b8 100755
--- a/scripts/build-goma-dependencies.sh
+++ b/scripts/build-goma-dependencies.sh
@@ -26,7 +26,7 @@
 
 # if CC_NAME="user" the following variables must be specified manually
 
-SYSTEM_COMPILER_BASE=/usr
+SYSTEM_COMPILER_BASE=/projects/sierra/linux_rh7/SDK/compilers/gcc/7.2.0-RHEL7
 SYSTEM_CC=$SYSTEM_COMPILER_BASE/bin/gcc
 SYSTEM_CXX=$SYSTEM_COMPILER_BASE/bin/g++
 SYSTEM_FC=$SYSTEM_COMPILER_BASE/bin/gfortran
@@ -58,11 +58,11 @@ FORTRAN_LIBS="-lgfortran"
 
 # Tells the TPL builder where MPI is installed.
 # Usually /usr/lib/openmpi/
-#export MPI_BASE_DIR=""
+export MPI_BASE_DIR=/projects/sierra/linux_rh7/SDK/compilers/gcc/7.2.0-RHEL7
 
 # Tells the TPL builder where the MPI compilers are.
 # Usually /usr/bin
-#export MPI_BIN_DIR=""
+export MPI_BIN_DIR=/projects/sierra/linux_rh7/SDK/compilers/gcc/7.2.0-RHEL7/bin
 
 
 # Tells the TPL builder which MPI you are using.
